python/cugraph/cugraph/structure/property_graph.py

```diff
@@ -849,7 +849,9 @@ def get_vertex_data(self, vertex_ids=None, types=None, columns=None):
                 # FIXME: invalid columns will result in a KeyError, should a
                 # check be done here and a more PG-specific error raised?
                 df = df[[self.type_col_name] + columns]
-            return df.reset_index()
+            df_out = df.reset_index()
+            df_out.index = df_out.index.astype(df.index.dtype)
```

Comment:
What does this do ?

Reply:
This ensures the dtype of the returned vertex ids matches the actual vertex id dtype.  Otherwise, there are issues encountered when passing returned vertex ids into cugraph algorithms because the datatype of the input vertices does not match the datatypes of the graph.  I added some explanatory comments and also cleaned up the code a bit.